### PR TITLE
add Nutanix CSI documentation

### PR DESCRIPTION
**Pull request by tuxtof**
**What this PR does / why we need it**:

Documentation on the different method to install Nutanix CSI Driver with CAPX clusters

**How Has This Been Tested?**:

by following the documentation and once deployed tested with the Nutanix CSI test framework

```release-note

Add Nutanix CSI Documentation

```

### CODE REVIEW

File: docs/install_csi_driver.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+
+## CAPI Workload cluster prerequisites for Nutanix CSI Driver
+
+Kubernetes workers need the following prerequisite to use the Nutanix CSI Drivers:
+
+- iSCSI initiator package (for Volumes based block storage)
+- NFS client package ( for Files based storage)
+
+These packages can already be present in the image you use with your Infrastructure provider or you can also rely on your bootstrap provider to install them.
```
Comment (deepakm-ntnx) [inline docs/install_csi_driver.md:16]: can we rephrase it without assumptions of package being present or user action is needed? how does one verify it?

File: docs/install_csi_driver.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+These `preKubeadmCommands` can be defined both in `KubeadmControlPlane` for master nodes and in `KubeadmConfigTemplate` for worker nodes.
+
+In our case with an Ubuntu 20.04 image, you need to modify both `KubeadmControlPlane` and `KubeadmConfigTemplate` as in the example below in order to install the prerequisites on all nodes.
+
+```yaml
+spec:
+  template:
+    spec:
+      # .......
+      preKubeadmCommands:
```
Comment (deepakm-ntnx) [inline docs/install_csi_driver.md:33]: can we make this part of the csi flavor of the cluster-template.yaml? say by creating cluster-template-nutanix-csi.yaml

File: docs/install_csi_driver.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    name: nutanix-csi-crs
+```
+
+The `clusterSelector` field controls how Cluster API will match this `ClusterResourceSet` on one or more workload clusters. In this case, I’m using a `matchLabels` approach where the `ClusterResourceSet` will be applied to all workload clusters having the `csi: nutanix` label present. If the label isn’t present, the `ClusterResourceSet` won’t apply to that workload cluster.
+
+The `resources` field references the ConfigMap created above, which contains the manifests for installing the Nutanix CSI driver.
+
+#### Assign the ClusterResourceSet to a workload cluster
+
+To assign this `ClusterResourceSet` to your workload cluster this is pretty simple, you just need to add the right label to your `Cluster` resource.
```
Comment (deepakm-ntnx) [inline docs/install_csi_driver.md:147]: "to your workload cluster ->this<- is pretty simple" => "to your workload cluster is pretty simple"

File: docs/install_csi_driver.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+You can do it a before a workload cluster creation by editing the output of the `clusterctl generate cluster` command or modify an already deployed workload cluster.
```
Comment (deepakm-ntnx) [inline docs/install_csi_driver.md:149]: extra "a" before "before" word needs to be removed from stmt

**Review by yannickstruyf3 — APPROVED** [review]

File: docs/install_csi_driver.md [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+
+
+
+## Install Nutanix CSI Driver with Helm
+
+The following procedure needs to be applied on a ready workload cluster.
+You can retrieve your workload cluster's kubeconfig and connect with the following command:
+
+```shell
+clusterctl get kubeconfig $CLUSTER_NAME -n $CLUSTER_NAMESPACE > $CLUSTER_NAME-KUBECONFIG
```
Comment (deepakm-ntnx) [inline docs/install_csi_driver.md:53]: should we provide sample values instead of env variables like $CLUSTER_NAME-KUBECONFIG etc

**Review by deepakm-ntnx — APPROVED** [review]

**Pull request merged**